pkg/antctl/raw/multicluster/join/join.go

```diff
@@ -0,0 +1,368 @@
+					fmt.Fprintf(cmd.OutOrStdout(), "Failed to delete ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, err)
+				} else {
+					fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim deleted \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, memberClusterNamespace)
+				}
+			}
+		}()
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ClusterSet \"%s\" in Namespace %s\n", memberClusterSet, memberClusterNamespace)
+	if createErr = k8sClient.Create(context.TODO(), newClusterSet(options.MemberClusterOpt.ClusterSet, memberClusterNamespace, options.LeaderClusterOpt.Server,
+		options.LeaderClusterOpt.Secret, memberClusterID, options.LeaderClusterOpt.ClusterID, options.LeaderClusterOpt.Namespace)); createErr != nil {
+		if apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "ClusterSet \"%s\" already exists in Namespace %s\n", memberClusterSet, memberClusterNamespace)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ClusterSet \"%s\": %v\n", memberClusterSet, createErr)
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterSet \"%s\" created in Namespace %s\n", memberClusterSet, memberClusterNamespace)
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for member cluster ready\n")
+	if err := waitForMemberClusterReady(cmd, k8sClient); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to wait for member cluster ready: %v\n", err)
+		return err
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Member cluster joined successfully\n")
+
+	return nil
+}
+
+func waitForMemberClusterReady(cmd *cobra.Command, k8sClient client.Client) error {
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for ClusterSet \"%s\" in Namespace %s to be registered in leader cluster\n",
+		options.MemberClusterOpt.ClusterSet, options.MemberClusterOpt.Namespace)
+	secret := &v1.Secret{}
+	if err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: options.LeaderClusterOpt.Secret, Namespace: options.MemberClusterOpt.Namespace}, secret); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to get secret \"%s\" in Namespace %s: %v\n", options.LeaderClusterOpt.Secret, options.MemberClusterOpt.Namespace, err)
+		return err
+	}
+
+	crtData, token, err := commonarea.GetSecretCACrtAndToken(secret)
+	if err != nil {
+		return err
+	}
+	config, err := clientcmd.BuildConfigFromFlags(options.LeaderClusterOpt.Server, "")
+	if err != nil {
+		return err
+	}
+	config.BearerToken = string(token)
+	config.CAData = crtData
+
+	remoteClient, err := client.New(config, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	memberClusterAnnounce := "member-announce-from-" + options.MemberClusterOpt.ClusterID
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for MemberClusterAnnounce \"%s\" in Namespace %s in leader cluster\n", memberClusterAnnounce, options.LeaderClusterOpt.Namespace)
```

Comment:
I feel there is no need to mention about `MemberClusterAnnounce`, it's too detail for user to understand. Maybe just say `Waiting for member cluster %s ready in Namespace...`